src/ng/http.js

```diff
@@ -1249,7 +1249,11 @@ function $HttpProvider() {
           cache,
           cachedResp,
           reqHeaders = config.headers,
-          url = buildUrl(config.url, config.paramSerializer(config.params));
+          // origUrl could be a $sce trusted URL which used a wrapper class
+          origUrl = config.url,
+          url = $sce.valueOf(origUrl);
```

Comment:
Isn't there a check near the top of `$http` that will fail if `requestConfig.url` is not a string?
That check also needs to be tweaked to accept `$sce`-wrapped values.

(I haven't looked at the failed tests, but I hope some failed because of this :smiley:)